docs/source/building/variants.rst

```diff
@@ -0,0 +1,1201 @@
+                                                  u'requirements': {u'build': [u'openssl 1.0.2k 0',
+                                                                               u'pip 9.0.1 py27_1',
+                                                                               u'python 2.7.13 0',
+                                                                               u'readline 6.2 2',
+                                                                               u'setuptools 27.2.0 py27_0',
+                                                                               u'sqlite 3.13.0 1',
+                                                                               u'tk 8.5.18 0',
+                                                                               u'wheel 0.29.0 py27_0',
+                                                                               u'zlib 1.2.8 3']},
+                                                  u'source': {u'path': u'/Users/msarahan/code/conda-build/tests/test-recipes/split-packages/_rm_rf_stays_within_prefix'}}}
+
+
+.. _extra_jinja2:
+
+Extra Jinja2 functions
+----------------------
+
+
+Two especially common operations when dealing with these API and ABI
+incompatibilities are ways of specifying such compatibility, and of explicitly
+expressing the compiler to be used. Three new Jinja2 functions are available when
+evaluating ``meta.yaml`` templates:
+
+* ``pin_compatible('package_name', min_pin='x.x.x.x.x.x', max_pin='x',
+  lower_bound=None, upper_bound=None)``: To be used as pin in run and/or test
+  requirements. Takes package name argument. Looks up compatibility of named
+  package installed in the build environment, and writes compatible range pin
+  for run and/or test requirements. Defaults to a semver-based assumption:
+  ``>=(current version),<(next major version)``. Pass ``min_pin`` or ``max_pin``
+  a `Pinning expressions`_ . This will be enhanced as time goes on with
+  information from `ABI Laboratory <https://abi-laboratory.pro/>`_
+
+* ``pin_subpackage('package_name', min_pin='x.x.x.x.x.x', max_pin='x',
+  exact=False)``: To be used as pin in run and/or test requirements. Takes
+  package name argument. Used to refer to particular versions of subpackages
+  built by parent recipe as dependencies elsewhere in that recipe. Can use
+  either pinning expressions, or exact (including build string).
+
+* ``compiler('language')``: To be used in build requirements most commonly.
+  Run or test as necessary. Takes language name argument. This is shorthand to
+  facilitate cross compiler usage. This Jinja2 function ties together two
+  variant variables, ``{language}_compiler`` and ``target_platform``, and
+  outputs a single compiler package name. For example, this could be used to
+  compile outputs targeting x86_64 and arm in one recipe, with a variant.
+
+There are default "native" compilers that and runtimes that are used when no
```

Comment:
You can drop "that" after compilers.  

This is the only occurrence of "runtimes" and I assume it is what you meant by "detailed further" in `.. _special_variant_keys`. 
1. It would be helpful to put runtimes in bactics so that it stands out like a key from conda_build_config.yaml when following the link from `.. _special_variant_keys`.
2. It would be nice to have a definition of what you consider `runtimes` and how they are different than compilers.